src/lib/componentUtils.js

```diff
@@ -0,0 +1,11 @@
+/* eslint import/prefer-default-export: 0 */
+/* This file will contain more functions in the future */
+
+/**
+ * Asserts that the passed Component is a React class component.
+ *
+ * @param {Function|Component} Component A Component to check
+ * @return {Object}
+ */
+export const isClassComponent = Component =>
+  typeof Component === 'function' && Component.prototype && !!Component.prototype.isReactComponent
```

Comment:
Perhaps we also add:

```jsx
export const isStatelessComponent = Component =>
  typeof Component === 'function' && Component.prototype && !Component.prototype.render

export const supportsRef = Component => 
  typeof Component === 'string' || isClassComponent(Component)
```

It may help clarify the factory function, see below.